src/m365/spo/commands/file/file-rename.ts

```diff
@@ -0,0 +1,161 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { urlUtil, validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  sourceUrl: string;
+  targetFileName: string;
+  force?: boolean;
+}
+
+class SpoFileRenameCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_RENAME;
+  }
+
+  public get description(): string {
+    return 'Renames a file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.force = !!args.options.force;
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const webUrl = args.options.webUrl;
+    const originalFileServerRelativeUrl: string = urlUtil.getServerRelativePath(args.options.webUrl, args.options.sourceUrl);
+    // Check if the source file exists.
+    // Called on purpose, we explicitly check if user specified file exists, otherwise, the rename can't happen and we will throw an error
+    this
+      .fileExists(originalFileServerRelativeUrl, webUrl)
+      .then((): Promise<void> => {
+        // Check if file already exists with the new name, if so: the file will be removed to the recycle bin
+        if (args.options.force) {
+          // delete the target file if force is set
```

Comment:
lets use comments only in places where it might be needed. Here the logic is simple and self explanatory. Also in other places as well